src/pycsou/opt/solver/pds.py

```diff
@@ -0,0 +1,965 @@
+            writeback_rate=writeback_rate,
+            verbosity=verbosity,
+            log_var=log_var,
+        )
+
+    @pycrt.enforce_precision(i=["x0", "z0", "tau", "sigma", "rho"], allow_None=True)
+    def m_init(
+        self,
+        x0: pyct.NDArray,
+        z0: typ.Optional[pyct.NDArray],
+        tau: typ.Optional[pyct.Real] = 1.0,
+        sigma: typ.Optional[pyct.Real] = None,
+        rho: typ.Optional[pyct.Real] = None,
+    ):
+        mst = self._mstate  # shorthand
+        mst["x"] = mst["x_prev"] = x0
+        mst["z"] = mst["z_prev"] = self.set_dual_variable(z0)
+        mst["tau"] = tau
+        mst["sigma"] = sigma
+        mst["rho"] = rho
+
+
+CPS = ChambollePockSplitting
+
+
+class DouglasRachfordSplitting(PDS):
+    r"""
+    Douglas Rachford splitting algorithm.
+
+    This class is also accessible via the alias ``DRS()``.
+
+    Notes
+    -----
+    The *Douglas Rachford Splitting (DRS)* can be used to solve problems of the form:
+
+    .. math::
+       {\min_{\mathbf{x}\in\mathbb{R}^N} \mathcal{G}(\mathbf{x})\;\;+\;\;\mathcal{H}(\mathbf{x}).}
+
+    where:
+
+    * :math:`\mathcal{G}:\mathbb{R}^N\rightarrow \mathbb{R}\cup\{+\infty\}` and :math:`\mathcal{H}:\mathbb{R}^M\rightarrow \mathbb{R}\cup\{+\infty\}` are two *proper*, *lower semicontinuous* and *convex functions* with *simple proximal operators*.
+    * The problem is *feasible* --i.e. there exists at least one solution.
+
+    **Remark 1:**
+    The algorithm is still valid if one of the terms :math:`\mathcal{G}` or :math:`\mathcal{H}` is zero.
+    **Default values of the hyperparameters provided here always ensure convergence of the algorithm.**
+
+    ``DouglasRachfordSplitting.fit()`` **Parameterization**
+
+    x0: NDArray
+        (..., N) initial point(s) for the primal variable.
+    z0: NDArray
+        (..., N) initial point(s) for the dual variable.
+        If None (default), then use x0 as the initial point(s) for the dual variable as well.
+    tau: Real
+        Primal step size
+
+    See Also
+    --------
+    :py:class:`~pycsou.opt.solver.pds.DRS`, :py:class:`~pycsou.opt.solver.pds.PrimalDualSplitting`, :py:class:`~pycsou.opt.solver.pds.ChambollePockSplitting`
+    """
+
+    def __init__(
+        self,
+        g: typ.Optional[pyco.ProxFunc] = None,
+        h: typ.Optional[pyco.ProxFunc] = None,
+        *,
+        folder: typ.Optional[pyct.PathLike] = None,
+        exist_ok: bool = False,
+        writeback_rate: typ.Optional[int] = None,
+        verbosity: int = 1,
+        log_var: pyct.VarName = ("x",),
+    ):
+        super(DouglasRachfordSplitting).__init__(
+            f=None,
+            g=g,
+            h=h,
+            K=None,
+            folder=folder,
+            exist_ok=exist_ok,
+            writeback_rate=writeback_rate,
+            verbosity=verbosity,
+            log_var=log_var,
+        )
+
+    @pycrt.enforce_precision(i=["x0", "z0", "tau", "sigma", "rho"], allow_None=True)
+    def m_init(
```

Comment:
Again we must set ``tau`` here in case the user does not provide it and make sure the algorithm converges.